src/Elastic.Apm/Logging/LogValuesFormatter.cs

```diff
@@ -46,19 +49,41 @@ public LogValuesFormatter(string format, string scope = null)
 					// Format item syntax : { index[,alignment][ :formatString] }.
 					var formatDelimiterIndex = FindIndexOfAny(format, FormatDelimiters, openBraceIndex, closeBraceIndex);
 
-					sb.Append(format, scanIndex, openBraceIndex - scanIndex + 1);
-					sb.Append((ValueNames.Count).ToString(CultureInfo.InvariantCulture));
-					ValueNames.Add(format.Substring(openBraceIndex + 1, formatDelimiterIndex - openBraceIndex - 1));
-					sb.Append(format, formatDelimiterIndex, closeBraceIndex - formatDelimiterIndex + 1);
+					if (ValueNames.Count < expectedArgs)
+					{
+						sb.Append(format, scanIndex, openBraceIndex - scanIndex + 1);
+						sb.Append(ValueNames.Count.ToString(CultureInfo.InvariantCulture));
+						ValueNames.Add(format.Substring(openBraceIndex + 1, formatDelimiterIndex - openBraceIndex - 1));
+						sb.Append(format, formatDelimiterIndex, closeBraceIndex - formatDelimiterIndex + 1);
+					}
+					else
+					{
+						_isArgumentNumberMismatch = true;
+						_mismatchedValueNames.Add(format.Substring(openBraceIndex + 1, formatDelimiterIndex - openBraceIndex - 1));
+					}
 
 					scanIndex = closeBraceIndex + 1;
 				}
 			}
 
+			if (_isArgumentNumberMismatch)
+			{
+				sb.AppendLine();
+				sb.AppendLine(
+					$"Above line is from an invalid structured log and may not be complete: number of arguments is not matching the number of placeholders, placeholders with missing values: {string.Join(", ", _mismatchedValueNames.ToArray())}");
+			}
+
+			if (ValueNames.Count != expectedArgs)
+			{
+				sb.AppendLine();
+				sb.AppendLine(
+					$"Above line is from an invalid structured log and may not be complete: number of placeholders in the log message does not match the number of parameters.");
```

Comment:
I would also appending 
1. How many arguments miss placeholders 
2. Arguments that miss placeholders. It can be very useful if the invalid structured log line came up for the first time while handling a customer support escalation and those arguments are very important for investigation. Otherwise we will have to release a build with the invalid structured log statement fixed.

Reply:
Yeah, that makes sense. I implemented this.